python/dgl/graphbolt/graph_storage/csc_sampling_graph.py

```diff
@@ -0,0 +1,267 @@
+            isinstance(x, str) for x in node_types
+        ), "Node type name should be string."
+        assert all(
+            isinstance(x, int) for x in node_type_ids
+        ), "Node type id should be int."
+        assert all(
+            isinstance(x, int) for x in edge_type_ids
+        ), "Edge type id should be int."
+        assert len(node_type_ids) == len(
+            set(node_type_ids)
+        ), "Multiple node types shoud not be mapped to a same id."
+        assert len(edge_type_ids) == len(
+            set(edge_type_ids)
+        ), "Multiple edge types shoud not be mapped to a same id."
+        edges = set()
+        for edge_type in edge_types:
+            src, edge, dst = edge_type
+            assert isinstance(edge, str), "Edge type name should be string."
+            assert edge not in edges, f"Edge type {edge} is defined repeatedly."
+            edges.add(edge)
+            assert (
+                src in node_types
+            ), f"Unrecognized node type {src} in edge type {edge_type}"
+            assert (
+                dst in node_types
+            ), f"Unrecognized node type {dst} in edge type {edge_type}"
+        self.node_type_to_id = node_type_to_id
+        self.edge_type_to_id = edge_type_to_id
+
+
+class CSCSamplingGraph:
+    r"""Class for CSC sampling graph."""
+
+    def __repr__(self):
+        return _csc_sampling_graph_str(self)
+
+    def __init__(
+        self, c_csc_graph: torch.ScriptObject, metadata: Optional[GraphMetadata]
+    ):
+        self.c_csc_graph = c_csc_graph
+        self.metadata = metadata
+
+    @property
+    def num_nodes(self) -> int:
+        """Returns the number of nodes in the graph.
+
+        Returns
+        -------
+        int
+            The number of rows in the dense format.
+        """
+        return self.c_csc_graph.num_nodes()
+
+    @property
+    def num_edges(self) -> int:
+        """Returns the number of edges in the graph.
+
+        Returns
+        -------
+        int
+            The number of edges in the graph.
+        """
+        return self.c_csc_graph.num_edges()
+
+    @property
+    def csc_indptr(self) -> torch.tensor:
+        """Returns the indices pointer in the CSC graph.
+
+        Returns
+        -------
+        torch.tensor
+            The indices pointer in the CSC graph. An integer tensor with
+            shape `(num_nodes+1,)`.
+        """
+        return self.c_csc_graph.csc_indptr()
+
+    @property
+    def indices(self) -> torch.tensor:
+        """Returns the indices in the CSC graph.
+
+        Returns
+        -------
+        torch.tensor
+            The indices in the CSC graph. An integer tensor with shape
+            `(num_edges,)`.
+
+        Notes
+        -------
+        It is assumed that edges of each node are already sorted by edge type
+        ids.
+        """
+        return self.c_csc_graph.indices()
+
+    @property
+    def node_type_to_id(self) -> Optional[Dict[str, int]]:
```

Comment:
Could consider just return the metadata object to users to save the effort of adding those attributes. Of course, you should document important attributes in the metadata class.

Reply:
+1 just return metadata.